### PR TITLE
Bug 1813416 - Clear FLAG_KEEP_SCREEN_ON when playing media is finished.

### PR DESCRIPTION
**Pull request by makotokato**
When media session is destroyed, we forget clearing ` FLAG_KEEP_SCREEN_ON`. So we should clear it when active session is nothing or not fullscreen.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1813416

### CODE REVIEW

File: android-components/components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt [hunk 1]
```diff
@@ -370,6 +371,8 @@ class MediaSessionFullscreenFeatureTest {
         store.waitUntilIdle()
         verify(activity.window).clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
 
+        clearInvocations(activity.window)
```
Comment (Mugurell) [inline android-components/components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:374]: Nice catch!
If we're resetting the `activity.window` state then the `atLeastOnce()` from below can probably be avoided also.

**Review by Mugurell — COMMENTED** [review]
LGTM though this would use some tests for when there are no active media tabs or when those are not in fullscreen anymore.
I can help also with this if the code is new to you.

File: android-components/components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt (hunk 1 above)
Comment (makotokato) [reply android-components/components/feature/media/src/test/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeatureTest.kt:374]: Thanks, I remove `atLeastOne`

**Comment by makotokato** [comment]
> I can help also with this if the code is new to you.

I add fullscreen enter/exit test and no active media test.  If I should change it, let me know.

**Review by Mugurell — APPROVED** [review]
Looks great! Thank you!

**Comment by makotokato** [comment]
@Mugurell I don't have any permission to land this. Could you land/merge this? (I don't know a-c landing rule)

**Comment by makotokato** [comment]
@mergifyio backport releases_v110

**Comment by rvandermeulen** [comment]
@Mergifyio backport releases_v110

**Pull request merged**